### DISCUSSION TITLE
Optimization

### DISCUSSION THREAD

**Discussion by mvn-luanvu-hn** [monstar-lab-oss/ml-frontend#10] [General]
- Reduce the bundle size
- Use Lighthouse to analyze

**Comment by yannde** [comment]
Should images, icons and fonts be part of this topic or in a separate one? (most choices are related to optimization)

**Reply by mvn-luanvu-hn** [reply]
Hi @yannde 
If images, icons, fonts need to be optimized, I think we can discuss here

**Reply by yannde** [reply]
- Images
  - png/jpg/webp
    - Responsive sizes
    - Progressive loading
  - svg
  - Background <> inline
  - Lazy load
- Icons
  - Library <> custom
  - svg <> icon font
- Fonts
  - Use font service (google / adobe)
  - If not using a font service
    - Use `unicode range` to serve smaller files in case of Japanese
    - Serve different formats ttf, woff, etc...
  - Preload fonts
